components/content-service/pkg/service/content-service.go

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package service
+
+import (
+	"context"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/gitpod-io/gitpod/common-go/tracing"
+	"github.com/gitpod-io/gitpod/content-service/api"
+	"github.com/gitpod-io/gitpod/content-service/pkg/storage"
+	"github.com/opentracing/opentracing-go"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ContentService implements ContentServiceServer
+type ContentService struct {
+	cfg storage.Config
+	s   storage.PresignedAccess
+}
+
+// NewContentService create a new content service
+func NewContentService(cfg storage.Config) (res *ContentService, err error) {
+	s, err := storage.NewPresignedAccess(&cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &ContentService{cfg, s}, nil
+}
+
+// DeleteUserContent deletes all content associated with a user.
+func (cs *ContentService) DeleteUserContent(ctx context.Context, req *api.DeleteUserContentRequest) (resp *api.DeleteUserContentResponse, err error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "DeleteUserContent")
+	span.SetTag("user", req.OwnerId)
+	defer tracing.FinishSpan(span, &err)
+
+	bucket := cs.s.Bucket(req.OwnerId)
+	err = cs.s.DeleteBucket(ctx, bucket)
+	if err == storage.ErrNotFound {
+		log.WithFields(log.OWI(req.OwnerId, "", "")).WithError(err).Error("DeleteUserContent failed: NotFound")
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
```

Comment:
Is this really an error case? If the user content is gone, it doesn't matter if it was us or someone else.
We could instead log a debug message and still return `DeleteUserContentResponse{}`

Reply:
Similar rationale as here: https://github.com/gitpod-io/gitpod/pull/3288#discussion_r584518536 But I'm not attached to it.

Reply:
I changed the behavior and do not return NotFound anymore.